tasks/badge.yaml

```diff
@@ -0,0 +1,375 @@
+
+          gh_group "🔍 Exemptions Validation"
+
+          ## Must minimize the scope and number of the exemptions to only what is absolutely required by the application
+
+          EXEMPTION_COUNT=$(kubectl get Exemptions -n "${NAMESPACE}" -o json | jq -r '.items | length')
+          if [ "$EXEMPTION_COUNT" -gt 0 ]; then
+            gh_warning "  ⚠️  Exemptions are present, review needed"
+          else
+            gh_notice "  ✅ No exemptions present"
+          fi
+
+          gh_endgroup
+
+          gh_group "🔍 Package Structure Validation"
+
+          ## Should expose all configuration (uds.dev CRs, additional Secrets/ConfigMaps, etc) through a Helm chart (ideally in a chart or charts directory).
+
+          if [ "${COMMON_ZARF}" == "true" ]; then
+            COMMON_ZARF_MANIFEST_COUNT=$(cat common/zarf.yaml | yq '.components[] | select(.manifests != null) | length' | wc -l)
+            if [ "$COMMON_ZARF_MANIFEST_COUNT" -gt 0 ]; then
+              gh_error "  ❌ Manifests present in common/zarf.yaml"
+            else
+              gh_notice "  ✅ No manifests present in common/zarf.yaml"
+            fi
+          fi
+
+          MAIN_ZARF_MANIFEST_COUNT=$(cat zarf.yaml | yq '.components[] | select(.manifests != null) | length' | wc -l)
+          if [ "$MAIN_ZARF_MANIFEST_COUNT" -gt 0 ]; then
+            gh_error "  ❌ Manifests present in zarf.yaml"
+          else
+            gh_notice "  ✅ No manifests present in zarf.yaml"
+          fi
+
+          ## Should implement or allow for multiple flavors (ideally with common definitions in a common directory)
+
+          ZARF_COMPONENTS_FLAVOR_COUNT=$(cat zarf.yaml | yq '.components[] | select(.only.flavor != null) | length' | wc -l)
+          if [ "$ZARF_COMPONENTS_FLAVOR_COUNT" -eq 0 ]; then
+            gh_error "  ❌ No flavors defined in zarf.yaml"
+          else
+            gh_notice "  ✅ At least one flavor defined in zarf.yaml"
+          fi
+
+          gh_endgroup
+
+          gh_group "🔍 Package Testing Validation"
+
+          ## Must implement Journey Testing to cover the basic user flows and features of the application, especially where an application interacts with an external service / interface.
+
+          if [ -d "tests" ]; then
+            if [ "$(ls -A tests)" ]; then
+              gh_notice "  ✅ Tests folder exists and has files"
+            else
+              gh_error "  ❌ Tests folder exists but is empty"
+            fi
+          else
+            gh_error "  ❌ Tests folder does not exist"
+          fi
+
+          gh_endgroup
+
+          gh_group "🔍 Versioning Validation"
+
+          # Must be consistently versioned across flavors - this can take many forms but flavors should differ in image bases/builds not application versions.
+
+          ZARF_PKG_VERSION=$(yq '.metadata.version' zarf.yaml | cut -d '-' -f1)
+          IMAGE_WITH_ZARF_VERSION_COUNT=$(cat zarf.yaml | yq '.components[].images' | grep -o "${ZARF_PKG_VERSION}" | wc -l)
```

Comment:
Another reason not to have this - this will fail on renovate tasks where temporarily the version of the package is different from that of the images - IMO package versioning is more of a release engineering process thing than an artifact of the package itself (i.e. you could lookup releases and validate that things matched but even then it is not always true: https://github.com/defenseunicorns/uds-package-sigstore/blob/main/zarf.yaml#L7

Reply:
While it will "fail" at least for now this is a tool for people to use to help with package dev ans verification, the ci never actually fails it just prints info

Reply:
I still think there's value if someone tries to version a package with the chart again for instance or makes a typo in the initial version

Reply:
If we wanted to make this fail in the future though this check wouldn't work still though - in the very least it should be only a warning IMO - some apps will need to ignore app versioning and will always "fail" and every app would see that failure on releases.  Since this is a callable workflow once it is enforced all of the downstream repos will start having issues once they pick it up.

Reply:
Enough apps include additional third party images versioned in their own scheme, even if if they version their app images the same as the chart.yaml appVersion. Given that i dont think this check really makes sense.

Reply:
changed to a warning for now